### DISCUSSION TITLE
How to use mandatory proxy of /a/b to /x/y

### DISCUSSION THREAD

**Discussion by Asilent** [Kong/kong#7400] [Help]
I want to let path /a/b redirect to service /x/y. Does Kong already support this action?

**Comment by kikito** [comment]
I am not 100% of what you are asking - in Kong, paths are managed by Routes while Services usually refer to "hosts". Could you give us a more specific example of what you want?

In any case Kong does not have a plugin that suits your needs, you can try with a community plugin. For example here's one redirect plugin from stone-payments:

https://github.com/stone-payments/kong-plugin-url-rewrite

**Comment by bungle** [comment]
I guess you mean to use: `HTTP` `Location` header with `301`, `302`, `303`, `307`, and `308` status codes?

redirection: send response to client to go another place
rewriting: change the uri before processing any further (e.g. before routing for example)

**Comment by Asilent** [comment]
What I described was a little confusing. Just like a request come from **www.x.com/a/b**, I want to send it to **www.y.com/c/d**. 
Now I find Kong can send a request like /a/b/c to /b/c by stripping path, but it can't meet my requirement.

**Comment by Asilent** [comment]
> I guess you mean to use: `HTTP` `Location` header with `301`, `302`, `303`, `307`, and `308` status codes?
> 
> redirection: send response to client to go another place
> rewriting: change the uri before processing any further (e.g. before routing for example)

No, just a proxy.  You can think that I want to anonymize real path, so I expose a virtual path to client. The real path is invisible to the user.

**Comment by kikito** [comment]
Kong by default is only able to "strip" parts or the full path of a request. In order to make a redirect to a different path you will need to use a plugin, like the one I linked on my first response, or else make a custom plugin which suits your needs.
